lib/network/modules/components/nodes/util/NodeBase.js

```diff
@@ -89,6 +90,23 @@ class NodeBase {
       }
     }
   }
+
+
+  /**
+   * Determine if the shape of a node needs to be recalculated.
+   *
+   * @protected
+   */
+  needRefresh(selected, hover) {
+    if (this.doRefresh) {
```

Comment:
It is a little dangerous to do a `if (this.doRefresh)` because this is valid e.g. also if doRefresh is a valid String or any object. That's why I would recommend to invert the logic:

```js
if (!this.doRefresh) { 
  return (this.width === undefined) || (this.labelModule.differentState(selected, hover));
}
this.doRefresh = false;
return true;
```